### ISSUE TITLE
clarify setup docs for developers

### ISSUE THREAD

**Issue by kstohr** [freedomofpress/securedrop#3403]
Update the  [Setting up the development environment](https://docs.securedrop.org/en/release-0.6/development/setup_development.html) doc to clarify a few setup issues: 

- highlight that the parent/project directory and a sub-directory are both named 'securedrop';  contributors need to cd into the sub_directory before running `make dev`  after installing the docker image.

- clarify that securedrop is built with Python 2.7.6. We are migrating to 3.0. (For example, the journalist app uses Python 3). 

- include links for options to install dependencies using homebrew.  

## User Research Evidence

These three issues were common stumbling blocks for contributors at the Pycon developer sprint.

**Comment by dachary** [comment]
@kstohr great feedback, thank you :-) The sub-directory problem is a FAQ, definitely :+1: 

Could you explain why explaining which python version is required is important. Is it so contributors refrain from using python3 constructs? That would make total sense, just checking if my understanding of the underlying problem is correct.

I'm not familiar with homebrew, in which context would it be useful?

**Comment by kstohr** [comment]
@dachary Yes.  Inspire folks to help migrate the codebase to Python 3, make them aware that some apps are in Python 2, some Python 3, and avoid python3 constructs in code that hasn't been migrated. 

[Homebrew](https://brew.sh/) is a commonly used package manager for Mac. In this case, you'd use it to manage the Vagrant, VirtualBox, Ansible... 

Also, I added a link to the [demo app]( http://demo.securedrop.club/) as that also came up yesterday.

**Comment by dachary** [comment]
>  Inspire folks to help migrate the codebase to Python 3, make them aware that some apps are in Python 2, some Python 3, and avoid python3 constructs in code that hasn't been migrated.

I'm very much in favor of bootstrapping such an effort. A good first step would be to add a documentation page with low hanging fruit to go in this direction?

**Comment by kstohr** [comment]
You could do that. Create a roadmap for the migration in the docs, and then
label git issues as ‘Python3’

>

**Comment by abhn** [comment]
> highlight that the parent/project directory and a sub-directory are both named 'securedrop'; contributors need to cd into the sub_directory before running make dev after installing the docker image.

Hey thanks for opening this. Just spent 20 minutes before landing on this issue and realizing my mistake.

**Comment by dachary** [comment]
@abhn This was [fixed recently](https://docs.securedrop.org/en/latest/development/setup_development.html#using-the-docker-environment) but maybe you used another version of the documentation?

**Comment by abhn** [comment]
Aah! Yes, that was the case. I was looking at stable. It is fixed in the latest.
